File: problems/1582-special-positions-in-a-binary-matrix/solution.py
class Solution:
    def numSpecial(self, mat: List[List[int]]) -> int:
        m = len(mat)
        n = len(mat[0])
        count = 0
        row_sums = [sum(row) for row in mat]
        col_sums = [sum(mat[i][j] for i in range(m)) for j in range(n)]
        for i in range(m):
            for j in range(n):
                if mat[i][j] == 1 and row_sums[i] == 1 and col_sums[j] == 1:
                    count += 1
        return count

File: problems/1582-special-positions-in-a-binary-matrix/solution.java
class Solution {
    public int numSpecial(int[][] mat) {
        int m = mat.length;
        int n = mat[0].length;
        int count = 0;

        for (int i = 0; i < m; i++) {
            for (int j = 0; j < n; j++) {
                if (mat[i][j] == 1) {
                    boolean rowSpecial = true;
                    for (int k = 0; k < n; k++) {
                        if (k != j && mat[i][k] == 1) {
                            rowSpecial = false;
                            break;
                        }
                    }

                    if (rowSpecial) {
                        boolean colSpecial = true;
                        for (int k = 0; k < m; k++) {
                            if (k != i && mat[k][j] == 1) {
                                colSpecial = false;
                                break;
                            }
                        }
                        if (colSpecial) {
                            count++;
                        }
                    }
                }
            }
        }

        return count;
    }
}